webhooks/virtualmachine/mutation/virtualmachine_mutator_hardware_controllers.go

```diff
@@ -231,8 +236,15 @@ func determineTargetController(
 
 	// If an existing controller does not exist with an available slot,
 	// create a new one if a bus is available.
-	for busNum := range controllerType.MaxCount() {
-		if _, exists := existingControllers[controllerType][busNum]; !exists {
+	startingBusNum := int32(0)
+	if !pkgcfg.FromContext(ctx).Features.AllDisksArePVCs {
+		// If all disks are PVCs is not enabled we need to skip bus number 0,
```

Comment:
I think Andrew also mentioned depends on this is not reliable. Might need to check with him whether this feature will always do registering
```
To be clear, we could also backfill them into spec for your feature too, but if we did that without actually registering them, it would cause the volume controller to forever spin
```